core/text/text_and_dictation.py

```diff
@@ -85,24 +85,15 @@ def prose_percent(m) -> str:
     return s + "%"
 
 
-@mod.capture(rule="<user.number_string> {user.currency_denomination}")
-def prose_simple_money(m) -> str:
-    return m.currency_denomination + m.number_string
-
-
-@mod.capture(
-    rule="<user.number_string> {user.currency_denomination} [and] <user.number_string> [cents|pence]"
-)
-def prose_money_with_cents(m) -> str:
-    return m.currency_denomination + m.number_string_1 + "." + m.number_string_2
-
-
-@mod.capture(rule="<user.prose_money_with_cents> | <user.prose_simple_money>")
+@mod.capture(rule="<user.number_string> {user.currency_denomination} [[and] <user.number_string> [cents|pence]]")
 def prose_money(m) -> str:
-    return str(m)
+    s = m.currency_denomination + m.number_string_1
+    if hasattr(m, "number_string_2"):
+        s += "." + m.number_string_2
+    return s
 
 
-hours_twelve = {
+hours_12 = {
```

Comment:
I also started with hours_12 and swear it caused parsing errors, which confused me since it obviously is a valid identifier, but when I changed to hours_twelve it worked.

Reply:
Note that I didn't change the name of the list talon-side, just this Python variable. Talon may not allow numbers in list names? Honestly not sure, I didn't try.